app/javascript/packs/shared/data_grid/DataGrid.jsx

```diff
@@ -0,0 +1,118 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import ReactDataGrid from 'react-data-grid';
+
+class DataGrid extends Component {
+  constructor(props) {
+    super(props);
+    this.cleanProps = this.cleanProps.bind(this);
+    this.transformColumns = this.transformColumns.bind(this);
+    this.setupDefaultProps = this.setupDefaultProps.bind(this);
+
+    this.transformColumns();
+    this.setupDefaultProps();
+
+    // Store the original rows array, and make a copy that
+    // can be used for modifying eg.filtering, sorting
+    this.state = {
```

Comment:
I don't understand first comment, it's already below `super(props)`. The second comment - I've just used the way the `react-data-grid` examples do it. If you have a better suggestion on how to do it, I'm all up for that.